moveit_ros/planning/kinematics_plugin_loader/src/kinematics_plugin_loader.cpp

```diff
@@ -324,41 +332,51 @@ robot_model::SolverAllocatorFn KinematicsPluginLoader::getLoaderFunction(const s
               }
             }
           }
 
-          std::string ksolver_timeout_param_name;
-          if (nh.searchParam(base_param_name + "/kinematics_solver_timeout", ksolver_timeout_param_name))
+          std::string ksolver_timeout_param_name = base_param_name + ".kinematics_solver_timeout";
+          if (ksolver_params->has_parameter(ksolver_timeout_param_name))
           {
             double ksolver_timeout;
-            if (nh.getParam(ksolver_timeout_param_name, ksolver_timeout))
+            rclcpp::Parameter parameter = node_->get_parameter(ksolver_timeout_param_name);
+            if (parameter.get_type() == rclcpp::ParameterType::PARAMETER_DOUBLE)
+            {
+              ksolver_timeout = ksolver_params->get_parameter<double>(ksolver_timeout_param_name);
               ik_timeout_[known_groups[i].name_] = ksolver_timeout;
-            else
+            }
+            else if (parameter.get_type() == rclcpp::ParameterType::PARAMETER_INTEGER)
             {  // just in case this is an int
               int ksolver_timeout_i;
-              if (nh.getParam(ksolver_timeout_param_name, ksolver_timeout_i))
-                ik_timeout_[known_groups[i].name_] = ksolver_timeout_i;
+              ksolver_timeout = ksolver_params->get_parameter<int>(ksolver_timeout_param_name);
+              ik_timeout_[known_groups[i].name_] = ksolver_timeout_i;
             }
           }
 
           // TODO: Remove in future release (deprecated in PR #1288, Jan-2019, Melodic)
           std::string ksolver_attempts_param_name;
-          if (nh.searchParam(base_param_name + "/kinematics_solver_attempts", ksolver_attempts_param_name))
+          if (ksolver_params->has_parameter(base_param_name + "/kinematics_solver_attempts"))
           {
-            ROS_WARN_ONCE_NAMED(LOGNAME, "Kinematics solver doesn't support #attempts anymore, but only a timeout.\n"
-                                         "Please remove the parameter '%s' from your configuration.",
-                                ksolver_attempts_param_name.c_str());
+            ksolver_attempts_param_name =
+                node_->get_parameter(base_param_name + "/kinematics_solver_attempts").get_value<std::string>();
+            RCLCPP_WARN(LOGGER_KINEMATICS_PLUGIN_LOADER,
+                        "Kinematics solver doesn't support #attempts anymore, but only a timeout.\n"
+                        "Please remove the parameter '%s' from your configuration.",
+                        ksolver_attempts_param_name.c_str());
           }
 
           std::string ksolver_res_param_name;
-          if (nh.searchParam(base_param_name + "/kinematics_solver_search_resolution", ksolver_res_param_name))
+          if (ksolver_params->has_parameter(base_param_name + "/kinematics_solver_search_resolution"))
           {
+            ksolver_res_param_name =
+                node_->get_parameter(base_param_name + "/kinematics_solver_search_resolution").get_value<std::string>();
```

Comment:
I'm pretty sure this doesn't work as intended. Here the parameter value is assigned to `ksolver_res_param_name` which is then again used as a key for parameter lookup. In the original implementation this made sense as `searchParam()` actually returns the parameter name. As described above, we can't do this here, so we just have to work with the result we got.